File: app/src/main/java/com/ab/ble/BleApplication.java
package com.ab.ble;

import android.app.Application;
import android.content.Context;

import com.ab.ble.data.local.SharedPrefsHelper;
import com.ab.ble.repository.SPRepository;

/**
 * Created by sunde_000 on 05/10/2017.
 */

public class BleApplication extends Application {
    private SPRepository SPRepository;

    @Override
    public void onCreate() {
        super.onCreate();
        Context context = getApplicationContext();
        SPRepository = new SPRepository(new SharedPrefsHelper(context));
    }

    public SPRepository getSPRepository() {
        return SPRepository;
    }

}

File: app/src/main/java/com/ab/ble/BleApplication.kt
package com.ab.ble

import android.app.Application
import com.ab.ble.data.local.SharedPrefsHelper
import com.ab.ble.repository.SPRepository

/**
 * Created by sunde_000 on 05/10/2017.
 */

class BleApplication : Application() {
    var spRepository: SPRepository? = null
        private set

    override fun onCreate() {
        super.onCreate()
        val context = applicationContext
        spRepository = SPRepository(SharedPrefsHelper(context))
    }

}